packages/grafana-e2e/src/flows/login.ts

```diff
@@ -3,20 +3,31 @@ import { e2e } from '../index';
 const DEFAULT_USERNAME = 'admin';
 const DEFAULT_PASSWORD = 'admin';
 
-export const login = (username = DEFAULT_USERNAME, password = DEFAULT_PASSWORD) => {
-  e2e().logToConsole('Logging in with username:', username);
-  e2e.pages.Login.visit();
-  e2e.pages.Login.username()
-    .should('be.visible') // prevents flakiness
-    .type(username);
-  e2e.pages.Login.password().type(password);
-  e2e.pages.Login.submit().click();
+export const login = (username = DEFAULT_USERNAME, password = DEFAULT_PASSWORD, loginViaApi = true) => {
+  if (loginViaApi) {
+    cy.request({
+      method: 'POST',
+      url: `${e2e.env('BASE_URL')}/login`,
+      body: {
+        user: username,
+        password,
+      },
+    });
+  } else {
+    e2e().logToConsole('Logging in with username:', username);
+    e2e.pages.Login.visit();
+    e2e.pages.Login.username()
+      .should('be.visible') // prevents flakiness
+      .type(username);
+    e2e.pages.Login.password().type(password);
+    e2e.pages.Login.submit().click();
 
-  // Local tests will have insecure credentials
-  if (password === DEFAULT_PASSWORD) {
-    e2e.pages.Login.skip().should('be.visible').click();
-  }
+    // Local tests will have insecure credentials
+    if (password === DEFAULT_PASSWORD) {
+      e2e.pages.Login.skip().should('be.visible').click();
+    }
```

Comment:
super minor nit, but i think code is easier to read when there's less nesting. 

Perhaps something like

```ts
if (!loginViaApi) {
    e2e().logToConsole('Logging in with username:', username);
    e2e.pages.Login.visit();
    e2e.pages.Login.username()
    etc
    return;
}

 cy.request({
      method: 'POST',
      url: `${e2e.env('BASE_URL')}/login`,
      body: {
        user: username,
        password,
      },
    });
```

or even like 

```ts
if (loginViaApi) {
  apiLoginStrategy()
} else {
  manualLoginStrategy()
```

dunno.

Reply:
I agree with @joshhunt on this, especially breaking the code up into 2 separate functions 👍